libsolidity/interface/StandardCompiler.cpp

```diff
@@ -850,15 +850,87 @@ Json::Value StandardCompiler::compileSolidity(StandardCompiler::InputsAndSetting
 	return output;
 }
 
+
+Json::Value StandardCompiler::compileYul(InputsAndSettings _inputsAndSettings)
+{
+	if (_inputsAndSettings.sources.size() != 1)
+		return formatFatalError("JSONError", "Yul mode only supports exactly one input file.");
+	if (!_inputsAndSettings.smtLib2Responses.empty())
+		return formatFatalError("JSONError", "Yul mode does not support smtlib2responses.");
+	if (!_inputsAndSettings.remappings.empty())
+		return formatFatalError("JSONError", "Field \"settings.remappings\" cannot be used for Yul.");
+	if (!_inputsAndSettings.libraries.empty())
+		return formatFatalError("JSONError", "Field \"settings.libraries\" cannot be used for Yul.");
+
+	Json::Value output;
+
+	AssemblyStack stack(_inputsAndSettings.evmVersion, AssemblyStack::Language::StrictAssembly);
+	string const& sourceName = _inputsAndSettings.sources.begin()->first;
+	string const& sourceContents = _inputsAndSettings.sources.begin()->second;
+	if (!stack.parseAndAnalyze(sourceName, sourceContents))
+	{
+		Json::Value errors = Json::arrayValue;
+		for (auto const& error: stack.errors())
+		{
+			auto err = dynamic_pointer_cast<Error const>(error);
+
+			errors.append(formatErrorWithException(
+				*error,
+				err->type() == Error::Type::Warning,
+				err->typeName(),
+				"general",
+				""
+			));
+		}
+		output["errors"] = errors;
+		return output;
+	}
+
+	if (isArtifactRequested(_inputsAndSettings.outputSelection, sourceName, "", "ast"))
+		// Does it make sense to output this as AST?
+		output["sources"][sourceName]["ast"] = stack.print();
+
+	if (_inputsAndSettings.optimiserSettings.runYulOptimiser)
+		stack.optimize();
+
+	MachineAssemblyObject object = stack.assemble(
+		AssemblyStack::Machine::EVM,
+		_inputsAndSettings.optimiserSettings.optimizeStackAllocation
+	);
+
+	string contractName = stack.parserResult()->name.str();
+
+	if (isArtifactRequested(
+		_inputsAndSettings.outputSelection,
+		sourceName,
+		contractName,
+		{ "evm.bytecode", "evm.bytecode.object", "evm.bytecode.opcodes", "evm.bytecode.sourceMap", "evm.bytecode.linkReferences" }
```

Comment:
Is `sourceMap` and `linkReferences` supported?

Reply:
It is supported by `collectEVMObject` - it is just not filled. I think it is better to keep it like that.